spark/src/main/java/org/apache/iceberg/spark/SparkReadOptions.java

```diff
@@ -33,6 +33,9 @@ private SparkReadOptions() {
   // A timestamp in milliseconds; the snapshot used will be the snapshot current at this time.
   public static final String AS_OF_TIMESTAMP = "as-of-timestamp";
 
+  // Whether to combine splits by partition boundary or not.
+  public static final String BY_PARTITION = "by-partition";
```

Comment:
It seems this is only used by `DataFrameReader` from Spark? if so we may need another way to config this.

Reply:
I think our common solution to this is to make a table property, I think in this use case it makes sense to have a table property and have a spark read option override for finer grained control